### PR TITLE
Complete the Cleanup for AST -> AstExpression Translation

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
This PR cleans up the work that has been merged to master over the last few PRs and also cleans up some functionality too.

### Important Notes
Parentheses in the syntax are now handled properly.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/main/scala/org/enso/compiler/generate/AstToAstExpression.scala [hunk 1]
```diff
@@ -219,30 +302,28 @@ object AstToAstExpression {
...
+    *
+    * @param callable the callable to translate
+    * @return the [[Core]] representation of `callable`
+    */
+  def translateCallable(callable: AST): AstExpression = {
+    callable match {
       case AstView.ForcedTerm(term) =>
-        AstForce(application.location, translateExpression(term))
+        AstForce(callable.location, translateExpression(term))
       case AstView.Application(name, args) =>
```
Comment (kustosz) [inline engine/runtime/src/main/scala/org/enso/compiler/generate/AstToAstExpression.scala]: Why is `Application` a part of `translateCallable`?

File: engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/MixfixFunctionsTest.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  subject should "easily support multiple arguments" in {
+    val code =
+      """
+        |type Foo a b
+        |
+        |Foo.if_then_else = a b -> case this of
+        |  Foo x y -> x + y + a + b
+        |
+        |if (Foo 1 2) then 3 else 4
```
Comment (kustosz) [inline engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/MixfixFunctionsTest.scala]: Is it not legal to leave the parens out?

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/thunk/CreateThunkNode.java [hunk 3]
```diff
@@ -5,18 +5,32 @@
 import org.enso.interpreter.node.ExpressionNode;
 import org.enso.interpreter.runtime.callable.argument.Thunk;
 
+/** THis node is responsible for wrapping a call target in a {@link Thunk} at execution time. */
```
[the remark this answers is not in the document]
Comment (iamrecursion) [reply engine/runtime/src/main/java/org/enso/interpreter/node/callable/thunk/CreateThunkNode.java]: Will go through and add these to all nodes when I get home.

File: engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/MixfixFunctionsTest.scala (hunk 2 above)
Comment (iamrecursion) [reply engine/runtime/src/test/scala/org/enso/interpreter/test/semantic/MixfixFunctionsTest.scala]: Seems to work fine! Changed one of the tests to not use parens.

**Pull request merged**